public/script.js

```diff
@@ -1582,39 +1580,26 @@ async function Generate(type, automatic_trigger, force_name2) {
 
         //////////////////////////////////
 
-        console.log('emptying chat2');
         let chat2 = [];
-        console.log('pre-replace chat.length = ' + chat.length);
-        for (let i = chat.length - 1, j = 0; i >= 0; i--, j++) {
-            let charName = selected_group ? chat[j].name : name2;
-            if (j == 0) {
-                chat[j]['mes'] = chat[j]['mes'].replace(/{{user}}/gi, name1);
-                chat[j]['mes'] = chat[j]['mes'].replace(/{{char}}/gi, charName);
-                chat[j]['mes'] = chat[j]['mes'].replace(/<USER>/gi, name1);
-                chat[j]['mes'] = chat[j]['mes'].replace(/<BOT>/gi, charName);
-            }
+        for (let i = coreChat.length - 1, j = 0; i >= 0; i--, j++) {
+            let charName = selected_group ? coreChat[j].name : name2;
             let this_mes_ch_name = '';
-            if (chat[j]['is_user']) {
+            if (coreChat[j]['is_user']) {
                 this_mes_ch_name = name1;
             } else {
                 this_mes_ch_name = charName;
             }
-            if (chat[j]['is_name']) {
-                chat2[i] = this_mes_ch_name + ': ' + chat[j]['mes'] + '\n';
+            if (coreChat[j]['is_name']) {
+                chat2[i] = this_mes_ch_name + ': ' + coreChat[j]['mes'] + '\n';
             } else {
-                chat2[i] = chat[j]['mes'] + '\n';
-            }
-            // system messages produce no text
-            if (chat[j]['is_system']) {
```

Comment:
System messages now filtered out not only for OpenAI, so this is no longer needed.